src/kv/kv.h

```diff
@@ -86,11 +87,22 @@ namespace kv
 
     struct LocalCommit
     {
+      LocalCommit() = default;
+      LocalCommit(Version v, State s, Write w) :
+        version(std::move(v)),
+        state(std::move(s)),
+        writes(std::move(w)),
+        next(nullptr),
+        prev(nullptr)
+      {}
+
       Version version;
       State state;
       Write writes;
+      LocalCommit* next;
+      LocalCommit* prev;
     };
-    using LocalCommits = std::list<LocalCommit>;
+    using LocalCommits = snmalloc::DLList<LocalCommit, std::nullptr_t, true>;
```

Comment:
I don't think `snmalloc` this should be hardcoded here? I wonder if it's worth having an abstraction layer so that we can choose `std::list` or `snmalloc::DLList` at compile time?

Reply:
dllist is an intrusive lined list vs an non intrusive linked list. The API is different so having a compile time flag is rather messy and I am not sure when you would want std::list